Formula/kubeless.rb

```diff
@@ -0,0 +1,74 @@
+class Kubeless < Formula
+  desc "Kubernetes Native Serverless Framework"
+  homepage "https://github.com/kubeless/kubeless"
+  url "https://github.com/kubeless/kubeless.git",
+      :tag => "v0.3.0",
+      :revision => "3103df24bccf67a71bef2ce792592d4ba9f09293"
+
+  depends_on "go" => :build
+  depends_on "kubernetes-cli" => :recommended
+
+  def install
+    commit = Utils.popen_read("git rev-parse --short HEAD").chomp
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/kubeless/kubeless").install buildpath.children
+    cd "src/github.com/kubeless/kubeless" do
+      ldflags = %W[
+        -w -X github.com/kubeless/kubeless/version.VERSION=v#{version}
+        -X github.com/kubeless/kubeless/version.GITCOMMIT=#{commit}
+      ]
+      system "go", "build", "-o", bin/"kubeless", "-ldflags", ldflags.join(" "),
+             "github.com/kubeless/kubeless/cmd/kubeless"
```

Comment:
Could this not simply be `"cmd/kubeless"` inside this directory, or?

Reply:
The goal is to install the binary in the `bin/` directory, isn't it? We could set the output of this command to any directory but then we would need to move it to its final destination.

Reply:
Yeah. What I mean is since we're already in `src/github.com/kubeless/kubeless` here why do we need to feed `go build` the full `github.com/kubeless/kubeless/cmd/kubeless` path instead of just `cmd/kubeless`?

Reply:
Or `./cmd/kubeless` perhaps, I can't remember off the top of my head which `go` insists is correct.

Reply:
ah, that's because the path should be relative to the GOPATH, inside the `src` directory. From `$GOPATH/src/` the correct relative path is that one. Ref: https://golang.org/doc/code.html#ImportPaths

Reply:
So `./cmd/kubeless` should work, no? 🙈

Reply:
Yeah, it does. My knowledge of `go` isn't completely shot, reassuringly, heh.

```diff
diff --git a/Formula/kubeless.rb b/Formula/kubeless.rb
index 852ddce8e1..018851a931 100644
--- a/Formula/kubeless.rb
+++ b/Formula/kubeless.rb
@@ -18,7 +18,7 @@ class Kubeless < Formula
         -X github.com/kubeless/kubeless/version.GITCOMMIT=#{commit}
       ]
       system "go", "build", "-o", bin/"kubeless", "-ldflags", ldflags.join(" "),
-             "github.com/kubeless/kubeless/cmd/kubeless"
+             "./cmd/kubeless"
       prefix.install_metafiles
     end
   end
```

```
==> Summary
🍺  /usr/local/Cellar/kubeless/0.3.0: 5 files, 27.9MB, built in 41 seconds
```

Reply:
True, that indeed works, I am just used to use the import path rather than current relative path (to make it work in any folder). In any case if that is a convention in this repo I am fine changing it.

Reply:
Makes sense to use the relative path here IMO, since Homebrew should always be building inside the `src/github.com/kubeless/kubeless` folder. Relative paths are the norm here when building from inside a directory, at least.